src/gpodder/download.py

```diff
@@ -936,14 +936,6 @@ def run(self):
             self.error_message = _('Error: %s') % (str(e),)
 
         with self:
-            if result == DownloadTask.FAILED:
```

Comment:
Is the moving of this block related to the fix or simply to make the normal path faster than the failure path?

Reply:
it was meant to make things easier to read by making the paths less complicated, I've further refactored the code so it is now:

if SUCCESS:
  do things
  return True

clean up after pause/cancel/fail

if FAIL:
  things
elif status = PAUSING:
  self.status = PAUSED
elif status = CANCELLING:
  self.status = CANCELLED

I think that's even clearer as there are less paths through the code.  (Happy to change this back if you thought it was clearer before)